# 1) A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions.


### H0(Null Hypothesis)= There is no significant difference between in the diameter of the cutlet between two units

### H1=(Alternative Hypothesis) There is significant difference between in the diameter of the cutlet between two units

In [2]:
import pandas as pd
import numpy as np
import scipy
from scipy import stats


In [3]:
Cutlets=pd.read_csv("C:\\Users\\Anjana\\Downloads\\Cutlets (1).csv")
Cutlets

Unit A  Unit B
0   6.8090  6.7703
1   6.4376  7.5093
2   6.9157  6.7300
3   7.3012  6.7878
4   7.4488  7.1522
5   7.3871  6.8110
6   6.8755  7.2212
7   7.0621  6.6606
8   6.6840  7.2402
9   6.8236  7.0503
10  7.3930  6.8810
11  7.5169  7.4059
12  6.9246  6.7652
13  6.9256  6.0380
14  6.5797  7.1581
15  6.8394  7.0240
16  6.5970  6.6672
17  7.2705  7.4314
18  7.2828  7.3070
19  7.3495  6.7478
20  6.9438  6.8889
21  7.1560  7.4220
22  6.5341  6.5217
23  7.2854  7.1688
24  6.9952  6.7594
25  6.8568  6.9399
26  7.2163  7.0133
27  6.6801  6.9182
28  6.9431  6.3346
29  7.0852  7.5459
30  6.7794  7.0992
31  7.2783  7.1180
32  7.1561  6.6965
33  7.3943  6.5780
34  6.9405  7.3875

In [4]:
stats.ttest_ind(Cutlets['Unit A'] ,Cutlets['Unit B']) #since p value>0.05 accept null hypothesis.

Ttest_indResult(statistic=0.7228688704678063, pvalue=0.4722394724599501)

### since the p valueis greater than 0.05, we reject the  Alternative hypothesis and accept Null Hypothesis
### H0(Null Hypothesis)= There is no significant difference between in the diameter of the cutlet between two units

# 2) A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch.
   
#  Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level.!

### H0(Null Hypothesis=There is no any difference in average TAT among the different laboratories.
### H1(Alternate Hypothesis=There is difference in average TAT among the difference laboratories

In [5]:
LabTAT=pd.read_csv("C:\\Users\\Anjana\\Downloads\\LabTAT (1).csv")
LabTAT

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0          185.35        165.53        176.70        166.13
1          170.49        185.91        198.45        160.79
2          192.77        194.92        201.23        185.18
3          177.33        183.00        199.61        176.42
4          193.41        169.57        204.63        152.60
..            ...           ...           ...           ...
115        178.49        170.66        193.80        172.68
116        176.08        183.98        215.25        177.64
117        202.48        174.54        203.99        170.27
118        182.40        197.18        194.52        150.87
119        182.09        215.17        221.49        162.21

[120 rows x 4 columns]

In [6]:
import scipy.stats as stats

In [7]:
stats.f_oneway(LabTAT.iloc[:,0],LabTAT.iloc[:,1],LabTAT.iloc[:,2],LabTAT.iloc[:,3])#p vale<0.05, accept alternate hypothesis

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

### Sine p value is very small ie, less than 5% we accept Alternative Hypothesis and reject null hypothesis
### H1(Alternate Hypothesis=There is difference in average TAT among the difference laboratories

# Question 3)

### H0(Null Hypothesis): All proportions are equal
### H1(Alternate Hypothesis): Not all proportions are equal

In [8]:
import pandas as pd

import scipy
from scipy import stats

In [9]:
BuyerRatio=pd.read_csv("C:\\Users\\Anjana\\Desktop\\ExcelR  DataScience Assignments\\Hypothesis testing\\BuyerRatio.csv")

In [10]:
BuyerRatio

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [11]:
stats.chi2_contingency(BuyerRatio.iloc[:,1:])

(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

In [12]:
stats.chi2_contingency(BuyerRatio.iloc[:,1:])[1]# pvalue>0.05, accept null hypothesis

0.6603094907091882

### Since p value is more than 5% we accept Null Hypothesis and reject alternative hypothesis
### H0(Null Hypothesis): All proportions are equal

# Q4)TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain %  of the customer order forms. Any error in order form renders it defective and has to be reworked before processing.  The manager wants to check whether the defective %  varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences


### H0=The defective percentage value in all the centers are equal
### H1=The defective percentage value in all centers are not equal

In [13]:
import pandas as pd

In [14]:
CustomerOrderForm=pd.read_csv("C:\\Users\\Anjana\\Desktop\\ExcelR  DataScience Assignments\\Hypothesis testing\\Costomer+OrderForm.csv")

In [15]:
CustomerOrderForm

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

In [16]:
Phillippines=pd.Series([len(CustomerOrderForm[CustomerOrderForm.Phillippines=='Error Free']),len(CustomerOrderForm[CustomerOrderForm.Phillippines=='Defective'])])

Indonesia=pd.Series([len(CustomerOrderForm[CustomerOrderForm.Indonesia=='Error Free']),len(CustomerOrderForm[CustomerOrderForm.Indonesia=='Defective'])])

Malta=pd.Series([len(CustomerOrderForm[CustomerOrderForm.Malta=='Error Free']),len(CustomerOrderForm[CustomerOrderForm.Malta=='Defective'])])

India=pd.Series([len(CustomerOrderForm[CustomerOrderForm.India=='Error Free']),len(CustomerOrderForm[CustomerOrderForm.India=='Defective'])])

In [17]:
CustomerOrderForm_2=pd.DataFrame()
CustomerOrderForm_2['Phillippines']=Phillippines
CustomerOrderForm_2['Indonesia']=Indonesia
CustomerOrderForm_2['Malta']=Malta
CustomerOrderForm_2['India']=India

In [18]:
CustomerOrderForm_2.set_axis(['Error Free','Detective'],inplace=True)

In [19]:
CustomerOrderForm_2

Phillippines  Indonesia  Malta  India
Error Free           271        267    269    280
Detective             29         33     31     20

In [20]:
stats.chi2_contingency(CustomerOrderForm_2)

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

### pvalue=0.2771020991233135 which is pvalue>0.05
### Null Hypothesis is accepted ,reject alternate hypothesis
### H0=The defective percentage value in all the centers are equal